weather_dl/download_pipeline/manifest.py

```diff
@@ -188,8 +188,8 @@ def _update(self, download_status: DownloadStatus) -> None:
 class GCSManifest(Manifest):
     """Writes a JSON representation of the manifest to GCS.
 
-    This is an append-only implementation, the latest value in the manifest
-    represents the current state of a download.
+    This is an overriding implementation, the latest value in the manifest
+    file represents the current state of a download.
```

Comment:
It would be better if we didn't delete the previous download status data in the manifest.

Reply:
GCS does not support append write mode, so the alternatives would be either:

- Keeping the GCS client open for writes, opening the door for timeouts, connection failures and unexpected behaviour if the updates span a long interval (which they typically do). I'm not sure if flushing works like regular POSIX files in GCS, so the latest manifest might not be available either while the downloads are in progress.
- Writing to multiple files and merging them in the end. As mentioned above I already have implemented this solution [here](https://github.com/blackvvine/weather-tools/blob/10dc5dd2a886c73580748efb94d04eb268caf61a/weather_dl/download_pipeline/manifest.py#L197) but it's a bit overly complex. (*)
- Some kind of hybrid between the two where we pool the output files but rotate them every X minutes (kinda like apache's `logrotate`). I couldn't find an out-of-the-box implementation for this so we'll have to build it from scratch, which might not be worth the effort.

(*) It can be simplified a *little* bit by not using `compose` for the merge, but that would probably make it less time-efficient.

Reply:
Here's an idea for another approach which contains tradeoffs: Assuming the lock works (this probably is not the case in a Dataflow environment), we could just open the manifest in read mode, copy it to local disk, append another row, and then write the data back to GCS. 

Given that this is a fragile area of code, a low performance option, and that we have plans for a better replacement (#60), I'm also happy to have the "fix" for this feature to simply be to remove the functionality. :)